docs/index.rst

```diff
@@ -106,23 +106,79 @@ You can use pip and make to quickly set up a development environment::
+        * In case any workaround is introduced, it is necessary to provide corresponding BZ ID directly into the code docstring
+        * All linting and formatting/style checks would be enforced by Travis-CI and PR would be considered broken until checks are passed successfully.
 * Adhere to typical commit guidelines:
-
-    * Commits should not cause NailGun to break.
+    * Commits should not cause NailGun’s unit test to fail. If it does, it will the responsibility of contributor to review those failures and fix them in the same PR's or raise another. The tracking of failures would be responsibility of contributor.
     * Commits should be small and coherent. One commit should address one issue.
     * Commits should have `good commit messages`_.
     * `Rebasing`_ is encouraged. Rebasing produces a much nicer commit history
       than merging.
+* To make the review process easy for all reviewers and anyone else interested in the new functionality, please provide some output making use of your changes. Having example of usage in docstring along with your code could really help others to build up on your code. You can add log from Python interactive shell or some tests results (from Robottelo / Foreman Ansible Modules) in PR message, or you can do something completely different - as long as it runs your code, it's fine!
+* If PR is applicable for many branches (e.g. master and one of '6.X.z' branches), specify that information in PR message
+* Unit tests
+    * Unit tests are compulsory
+    * Unit tests should cover all available actions, for the entity. For eg: Repository Sets, have enable, disable, list_available there should be unit tests exercising these actions.
+* When in doubt, ask on IRC. Join #robottelo on `freenode`_
+
+**Important to Note :**
+
+* Define Foreman Version labels in Nailgun
+    if possible, the contributor should set the right version.
+
+* All PRs should be raised along with Unit tests
+    The unit tests should be added while adding a new entity or modifying the existing entity or modifying and adding to the core of Nailgun.
+
+* Test results from upstream devel or from upstream nightly
+    The API call results are required to make the review process more firm. The PR author can provide results from any library that uses the contributed code by running the changes on upstream nightly or from his/her devel box. The interactive python shell output would be acceptable
+
+Nailgun Review Process
+______________________
+
+* Travis CI is run, and any issues are resolved by contributor.
+* If deemed necessary by contributors/reviewers, an automation run is triggered.
+* At least two ACKs are required to merge a pull request.
+* At least one ACK must be from a Tier 2 reviewer.
+* If a PR requires changes to the CI environment, the “CI Owner” must also provide an ACK.
+* Pull request is merged by the person who does the final required ACK.
+
```

Comment:
that is logical, but... `Pull request can be merged only when all comments are in resolved state`